lib/iolib/file.go

```diff
@@ -37,6 +38,9 @@ type File struct {
 	status fileStatus
 	reader bufReader
 	writer bufWriter
+	cmd *exec.Cmd
```

Comment:
If I understand correctly, `cmd` is only used when closing the file or when getting the file name and `stdin` and `stdout` are only used when closing the file.  I think it would be cleaner to have something like a `name` and a `close` fields, one can be used to get the file name if it's not a regular file, the other can be used to close the file (so all the logic can go into that callback).

(not a requirement to merge the PR though)